### PR TITLE
add all necessary files to start my first endpoint. Create first func…

### PR DESCRIPTION
**Pull request by aloush-dev**
added all the necessary files to begin coding my first endpoint.
Created my first get request endpoint.
made one tests 
made on error handling test

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      .get("/api/topics")
+      .expect(200)
+      .then((data) => {
+        data.body.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic).toHaveProperty("description");
+        });
+      });
+  });
+  test("400 : should respond with bad request if given an endpoint other than topics", () => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: wrong status code is description

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  return db.end();
+});
+
+describe("get api topics", () => {
+  test("200 : should respond with an array of topic objects with the properties slug and description", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((data) => {
+        data.body.forEach((topic) => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: what's to say you're responding with an empty array here? How many topics should come back?

File: controllers/get-controller.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const { fetchTopics } = require("../models/get-models");
+
+exports.getTopics = (req, res, next) => {
+  fetchTopics()
+    .then((data) => {
+      res.status(200).send(data);
```
Comment (daviidbartlett) [inline controllers/get-controller.js]: can we send  stuff back in an object please? eg. {topics: [...]}

File: controllers/get-controller.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    })
+    .catch((err)=>{
+        console.log(err)
```
Comment (daviidbartlett) [inline controllers/get-controller.js:9]: remove console.log

File: app.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const app = express();
+
+app.use(express.json());
```
Comment (daviidbartlett) [inline app.js]: why is this here?

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          expect(topic).toHaveProperty("description");
+        });
+      });
+  });
+  test("400 : should respond with bad request if given an endpoint other than topics", () => {
+    return request(app)
+      .get("/api/fish")
+      .expect(404)
+      .then((data) => {
+          console.log(data)
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: remove console.log

File: __tests__/app.test.js (hunk 1 above)
Comment (aloush-dev) [reply __tests__/app.test.js]: Changed to correct status code.

File: __tests__/app.test.js (hunk 2 above)
Comment (aloush-dev) [reply __tests__/app.test.js]: Added a .toHaveLength to check how many results are returned

File: controllers/get-controller.js (hunk 3 above)
Comment (aloush-dev) [reply controllers/get-controller.js]: Change made, data is sent back as an object

File: controllers/get-controller.js (hunk 4 above)
Comment (aloush-dev) [reply controllers/get-controller.js:9]: removed

File: __tests__/app.test.js (hunk 6 above)
Comment (aloush-dev) [reply __tests__/app.test.js]: removed

File: app.js (hunk 5 above)
Comment (aloush-dev) [reply app.js]: removed not needed yet
